business_calendar.go

```diff
@@ -127,3 +123,14 @@ func (c businessCalendar) closestBusinessDay(origin date.Date, forwards bool) da
 		}
 	}
 }
+
+// addWeekendDays adds or removes 2 days to the day count (nbDaysLeft) to skip the weekend days.
+func addWeekendDays(current date.Date, nbDaysLeft, days int) int {
+	if days > 0 && int(time.Friday-current.Weekday()) < nbDaysLeft {
+		return nbDaysLeft + 2
+	} else if days < 0 && int(current.Weekday()-time.Monday) < -nbDaysLeft {
+		return nbDaysLeft - 2
+	}
```

Comment:
With early return you don't need the else part at all (avoid it in general)
```suggestion
	if days > 0 && int(time.Friday-current.Weekday()) < nbDaysLeft {
		return nbDaysLeft + 2
	}
	
	if days < 0 && int(current.Weekday()-time.Monday) < -nbDaysLeft {
		return nbDaysLeft - 2
	}
```